src/backend/c-lisp.py

```diff
@@ -95,6 +112,35 @@ def gen_stmt(self, stmt):
             print(f"Error in statement: {stmt}")
             raise e
 
+    def is_while_stmt(self, stmt):
+        return stmt[0] == "while"
+
+    def gen_while_stmt(self, stmt):
+        if len(stmt) < 3:
+            raise CodegenError(f"Bad while statement: {stmt}")
+
+        cond_sym, loop_lbl, cont_lbl, break_lbl = [
+            random_label(CLISP_PREFIX, [extra])
```

Comment:
use scopes for all intermediate variables

Reply:
Done

Reply:
decided against it after looking at the outputs